### DISCUSSION TITLE
Using WAGMI with SSR (Next.js)

### DISCUSSION THREAD

**Discussion by smhutch** [wevm/wagmi#498] [Q&A]
After bumping `wagmi` to `0.3.5` in our project, we started to face a number of styling issues (here's an [example](https://twitter.com/0tl_/status/1527969108192800768?s=21&t=YsDlIghvhuvilfCQDBmbkg) 😆) due to a mismatch between the server-rendered HTML and the first render on the client side.

After some investigation, I discovered that this was due to hooks like `useAccount` returning `isLoading` as `true` during SSR, but as `false` on the client. Here's an example of the return value of `useAccount` during SSR and on the client:

![CleanShot 2022-05-24 at 20 32 02@2x](https://user-images.githubusercontent.com/7217244/170117859-9dbd9f7a-8b2c-4299-9550-3fdd583fa1f8.png)

Before we upgraded to `0.3`, the SSR and client output was consistent on first render. In this case it returned:

```tsx
[
  {
    "loading": false
  },
  null
]
```

A few questions: 

- Why is loading `true` on the server side. In my tests, it's also true when `autoConnect` is set to `false`?
- Is there a recommended pattern for handling SSR in wagmi? Currently we're manually patching this issue in many places, but I would prefer to help with a fix in Wagmi 😄 

I appreciate that this isn't necessarily a bug, but more of a side-effect that's having a knock-on effect at our end. I would guess that anyone using Next.js + WAGMI + [Stitches](https://stitches.dev/) will face a similar issue to us.

**Comment by smhutch** [comment]
This is an [repro](https://github.com/smhutch/wagmi-sandbox/blob/main/README.md) of the issue we're facing.
Styling bug isn't visible here, but mis-match between client/server output is highlighted in a console error.

**Reply by kris10cabrera** [reply]
copy & pasted content here to an issue -> https://github.com/tmm/wagmi/issues/542
